### PR TITLE
Fileinfo order 4

### PR DESCRIPTION
**Pull request by regit**
Update of #3344 implementing the change discussed in the previous PR.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- Update the logger APIs
- Fix ftp-data file extraction that was not respecting the client/server info

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/398
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/181

### CODE REVIEW

File: src/output-json-file.c [hunk 1]
```diff
@@ -80,10 +80,24 @@ typedef struct JsonFileLogThread_ {
...
+            fdir = LOG_DIR_FLOW_TOCLIENT;
+            break;
+        case STREAM_TOSERVER:
+            fdir = LOG_DIR_FLOW_TOSERVER;
+            break;
+        default:
+            fdir = LOG_DIR_FLOW;
+    }
+
+    js = CreateJSONHeader(p, fdir, "fileinfo");
```
Comment (victorjulien) [inline src/output-json-file.c]: declare js here again?

File: src/output-json-file.c [hunk 2]
```diff
@@ -80,10 +80,24 @@ typedef struct JsonFileLogThread_ {
...
+
+    switch(dir) {
+        case STREAM_TOCLIENT:
+            fdir = LOG_DIR_FLOW_TOCLIENT;
+            break;
+        case STREAM_TOSERVER:
+            fdir = LOG_DIR_FLOW_TOSERVER;
+            break;
+        default:
+            fdir = LOG_DIR_FLOW;
```
Comment (victorjulien) [inline src/output-json-file.c]: could this be set at 'fdir' declaration?
Might be good to have a DEBUG_VALIDATION_BUG_ON here too, as this default should be unreachable

File: src/output-json.c [hunk 3]
```diff
@@ -445,6 +445,68 @@ void JsonFiveTuple(const Packet *p, enum OutputJsonLogDirection dir, json_t *js)
             sp = p->sp;
             dp = p->dp;
             break;
+        case LOG_DIR_FLOW_TOSERVER:
```
Comment (victorjulien) [inline src/output-json.c]: is this the same as LOG_DIR_FLOW?

Comment (regit) [reply src/output-json.c]: Indeed

**Comment by regit** [comment]
Replace by #3349

**Pull request closed without merging**